packages/react-dom/src/__tests__/ReactDOMFizzServer-test.js

```diff
@@ -3732,9 +3734,15 @@ describe('ReactDOMFizzServer', () => {
       renderToPipeableStream(<App />).pipe(writable);
     });
 
+    await act(() => {
+      resolve();
+    });
+
     expect(getVisibleChildren(document)).toEqual(
       <html>
-        <head />
+        <head>
+          <link rel="preconnect" href="foo" />
+        </head>
```

Comment:
This demonstrates a small semantic difference in this PR. the foo preconnect resolves in a microtask which now runs before the flush. I updated the test to perform an additional preconnect new a new macrotask and it demonstrates that there is no attempted write to the stream after the render has completed.